libmozdata/utils.py

```diff
@@ -315,3 +316,32 @@ def get_params_for_url(params):
 
 def get_url(url):
     return url if url.endswith('/') else url + '/'
+
+
+def get_language(path):
+    """
+    Detect programming language or tool
+    from a filename
+    """
+    name = os.path.basename(path)
+    extension = os.path.splitext(name)[1][1:]
+    langs = {
+        'Shell': ['sh'],
+        'Xml': ['xml', 'xst'],
+        'Html': ['html', 'xhtml'],
+        'Css': ['css'],
+        'Javascript': ['js', 'jsm'],
+        'Makefile': ['mk', 'Makefile', 'Makefile.am', 'Makefile.in', 'configure.in', 'autoconf.mk.in'],
```

Comment:
We should also add `moz.build` files, I'm not sure how to categorize them.
Perhaps:
```
'moz.build': ['moz.build']
```